### DISCUSSION TITLE
Adding a custom button to Source Control view in extension

### DISCUSSION THREAD

**Discussion by vex9z7** [microsoft/vscode-discussions#487] [Extension Development QnA] [answered]
I'm developing a VS Code extension that adds a button to the Source Control view.
I've tried to follow the approach used in the Git extension's [actionButton.ts](https://github.com/microsoft/vscode/blob/main/extensions/git/src/actionButton.ts) file, but my button isn't showing up.
Here is the relevant code fragment.
``` 
const commitButton = {
      command: {
        title: 'Commit',
        command: 'my-commit-command',
      },
      description: 'Commit',
      enabled: true,
    };

    this._scm.commitButton = commitButton;
```
I suspect there might be some missing code or a problem with the API integration. If anyone has experience with this or any suggestions, I would greatly appreciate your help

**Comment by isidorn** [comment, marked as answer]
Not sure if it is possible. But in case it is you need to use the Menu contribution point https://code.visualstudio.com/api/references/contribution-points#contributes.menus

Thus you need to contribute a command to one of the SCM menus.

**Comment by r3m0t** [comment]
Check the proposed APIs used by the Git extension-

https://github.com/microsoft/vscode/blob/main/extensions/git/package.json#L12

https://github.com/microsoft/vscode/tree/main/src/vscode-dts
